lib/node.js

```diff
@@ -273,7 +273,29 @@ class Node {
 
     this.sockets.sub.connect(node.pub);
     this.pool[node.pub] = node;
-    this._heartbeat(node);
+    this._onHeartbeat(node);
+  }
+
+  _heartbeat () {
+    this.broadcast('cluster:heartbeat', this);
+
+    this.redis.smembers('cluster:discovery')
+      .then(members => {
+        let found = false;
+
+        for (const serialized of members) {
+          const member = JSON.parse(serialized);
+
+          if (member.router === this.config.bindings.router.href) {
+            found = true;
+            break;
+          }
+        }
+
+        if (!found) {
+          return this.join();
```

Comment:
What if joining fails? Shouldn't the node stop accepting requests and enter a reconnection loop, or something?

Reply:
We already have a loop. This code is triggered on each heartbeat, every 5s by default.
About rejecting requests, ideallly the cluster health check should  be KO till the node is not joined (which may not be the case if we have a 2n cluster setup and an actual split brain).
If the health check is down, the reverse proxy should be configured to exclude the node from the pool.